base_import_match/models/base.py

```diff
@@ -35,6 +35,7 @@ def load(self, fields, data):
                 if xmlid:
                     # Skip rows with ID, they do not need all this
                     row["id"] = xmlid
+                    newdata.append(tuple(row[f] for f in clean_fields))
                     continue
```

Comment:
It is added to the newdata list otherwise it won't affect any record even if it is found by external_id